### PR TITLE
Feature/BAIP-464-optional-fields-measure-input

### PR DESCRIPTION
**Pull request by bennyvdhoogen**

### CODE REVIEW

File: src/schema/measure/measure.factory.ts [hunk 1]
```diff
@@ -37,10 +37,10 @@ export class MeasureFactory {
 		measure.unitPrice = unitPrice;
 		measure.quantity = quantity;
 		measure.quantityUnitOfMeasurement = QuantityUnitOfMeasurement[quantityUnitOfMeasurement];
-		measure.surveyScopeId = surveyScopeId;
-		measure.failureModeId = failureModeId;
-		measure.defectId = defectId;
-		measure.manifestationId = manifestationId;
+		measure.surveyScopeId = surveyScopeId ?? null;
```
Comment (fortunato) [inline src/schema/measure/measure.factory.ts:40]: Doesn't Prisma already return null for these?

File: src/schema/measure/measure.repository.ts [hunk 2]
```diff
@@ -57,9 +43,37 @@ export class MeasureRepository implements IMeasureRepository {
 			description,
 			location,
 			quantity,
-			surveyScopeId,
+			surveyScopeId: surveyScopeId ?? null,
 		};
 
+		if (remarks) {
```
Comment (fortunato) [inline src/schema/measure/measure.repository.ts:49]: Could've done without the condition here. I believe Prisma either returns null or a string.

File: src/schema/measure/measure.repository.ts [hunk 3]
```diff
@@ -97,6 +112,10 @@ export class MeasureRepository implements IMeasureRepository {
 			surveyScopeId,
 		};
 
+		if (remarks) {
```
Comment (fortunato) [inline src/schema/measure/measure.repository.ts:115]: Could've done without the condition here. I believe Prisma either returns null or a string.

**Review by fortunato — CHANGES_REQUESTED** [review]

File: src/schema/measure/measure.repository.ts (hunk 2 above)
Comment (bennyvdhoogen) [reply src/schema/measure/measure.repository.ts:49]: I believe by default it does indeed return an empty string, which does not play nice with the isUUID decorator. There probably is a more concise way to do this, but I wouldn't know how.

**Comment by bennyvdhoogen** [comment]
> I believe by default it does indeed return an empty string, which does not play nice with the isUUID decorator. There probably is a more concise way to do this, but I wouldn't know how. 

Nope, this is nonsense. You were right.

File: src/schema/measure/measure.repository.ts (hunk 2 above)
Comment (bennyvdhoogen) [reply src/schema/measure/measure.repository.ts]: Nope, this is nonsense. You were right.

**Review by anaim — APPROVED** [review]

**Pull request merged**